Microsoft/BotBuilder-Samples
# Updates project templates to use latest AspNetCore references and code,

Fixes https://github.com/Microsoft/botbuilder-dotnet/issues/1478 and https://github.com/Microsoft/botbuilder-dotnet/issues/1421

## Proposed Changes

- Changed references to Microsoft.AspNetCore.All to Microsoft.AspNetCore.App in CoreBot.csproj, EchoBotWithCounter.csproj and EmptyBot.csproj
- Removed call to logging.AddAzureWebAppDiagnostics(); from Program.cs and added explicit reference to Microsoft.Extensions.Logging.AzureAppServices as documented [here](https://docs.microsoft.com/en-us/aspnet/core/fundamentals/logging/?view=aspnetcore-2.2#logging-in-azure)
- Updated Program.cs code to match OOTB AspNetCore web api template.
- Removed references for Microsoft.Azure.CognitiveServices.ContentModerator, Microsoft.Azure.CognitiveServices.Language preview and Microsoft.Graph from CoreBot.csproj


## Testing
1. Create a new bot project based on the template.
2. Upgrade BotBuilder nuget packages to 4.3
3. Add an instance for a variable of type CloudStorageAccount anywhere in the code.
4. The solution should compile.

## Additional Info
See https://docs.microsoft.com/en-us/aspnet/core/fundamentals/metapackage?view=aspnetcore-2.2 for additional details justifying the change based on new AspNetCore 2.1 recommendations.
